alc/backends/jack.cpp

```diff
@@ -508,15 +508,32 @@ std::string JackBackendFactory::probe(BackendType type)
         if(client == nullptr)
             throw al::backend_exception{ALC_INVALID_VALUE, "Failed to open client connection: 0x%02x",
                     status};
-        const char **ports{jack_get_ports(client, nullptr, nullptr, JackPortIsInput)};
+        const char **ports{jack_get_ports(client, nullptr, nullptr, JackPortIsPhysical|JackPortIsInput)};
         for (int i = 0; ports && ports[i] ; i++) {
+            const char* colon = strchr(ports[i], ':');
+            int len = static_cast<int>  (colon - ports[i]);
+            if (outnames.find(ports[i], 0, len) == std::string::npos) {
+                outnames.append(ports[i], len);
+                outnames.append(1, '\0');
+            }
+        }
+        jack_free(ports);
+        jack_client_close (client);
+
+        char *customDevices = strdup(GetConfigValue(nullptr, "jack", "custom-devices", ""));
+        size_t len = strlen(customDevices);
+        if (len != 0) {
+            char *ptr = strrchr(customDevices, ';');
+            while (ptr) {
+                *ptr = '\0';
+                ptr = strrchr(ptr, ';');
+            }
```

Comment:
This would need to separate the display name from the port name pattern. It will also need to maintain a list of custom names to know what the desired pattern is when such a name is given for the backend `open` call. And it probably should ensure there are some ports matching the pattern before adding it to the enumerated list.

Reply:
>  And it probably should ensure there are some ports matching the pattern before adding it to the enumerated list.

I think it would be better to let everything in the device list. And in the case of no ports corresponding, openal should just be let unconnected. Because if the user want to start a client which is in the `custom-devices` list after the `probe` call, it will be easier for the user to rechange the device choice than trying to figure out how to make the application retrigger `probe` to update the list (reopening the menu ? restarting the whole application ?). I don't think every application using openal would thinks that a device may appear during the run of the application, and make it difficult to update the list, as a device is unlikely to appear for Alsa/OSS/Pulseaudio/... but not for Jack.